File: leetcode/integer division/2894. Divisible and Non-divisible Sums Difference/python/main.py
class Solution:
    def differenceOfSums(self, n: int, m: int) -> int:
        end, ans = n + 1, 0

        for i in range(end):
            if i % m:
                ans += i
            else:
                ans -= i

        return ans


print(Solution().differenceOfSums(10, 3))

File: leetcode/integer division/2894. Divisible and Non-divisible Sums Difference/golang/main.go
package main

import (
	"fmt"
)

func differenceOfSums(n int, m int) int {
	ans := 0

	for i := 0; i <= n; i++ {
		if i%m > 0 {
			ans += i
		} else {
			ans -= i
		}
	}
	return ans
}
func main() {
	fmt.Println(differenceOfSums(10, 3))
}
